src/protocol/clock/extentmaker.rs

```diff
@@ -0,0 +1,88 @@
+use std::num::TryFromIntError;
+use std::time::Duration;
+
+use super::extent::{Extent, ExtentIncrement, ExtentIncrementsCount, IncrementCounter};
+use super::{ClockType, StoppedClock, TimeNow, WorkingClock};
+
+pub trait ExtentFromClock<T>: Sized
+where
+    T: TimeNow,
+{
+    fn now(size: &ExtentIncrement) -> Option<Result<Extent, TryFromIntError>> {
+        T::now()
+            .as_nanos()
+            .checked_div((*size).as_nanos())
+            .map(|quantity| match ExtentIncrementsCount::try_from(quantity) {
+                Err(error) => Err(error),
+                Ok(quantity) => Ok(Extent::new(size, &quantity)),
+            })
+    }
+
+    fn now_add(size: &ExtentIncrement, add_time: &Duration) -> Option<Result<Extent, TryFromIntError>> {
+        match T::add(add_time) {
+            None => None,
+            Some(time) => {
+                time.as_nanos()
+                    .checked_div(size.as_nanos())
+                    .map(|quantity| match ExtentIncrementsCount::try_from(quantity) {
+                        Err(error) => Err(error),
+                        Ok(quantity) => Ok(Extent::new(size, &quantity)),
+                    })
+            }
+        }
+    }
+    fn now_sub(size: &ExtentIncrement, sub_time: &Duration) -> Option<Result<Extent, TryFromIntError>> {
+        match T::sub(sub_time) {
+            None => None,
+            Some(time) => {
+                time.as_nanos()
+                    .checked_div(size.as_nanos())
+                    .map(|quantity| match ExtentIncrementsCount::try_from(quantity) {
+                        Err(error) => Err(error),
+                        Ok(quantity) => Ok(Extent::new(size, &quantity)),
+                    })
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ExtentClock<const T: usize> {}
+
+pub type WorkingClockExtentMaker = ExtentClock<{ ClockType::WorkingClock as usize }>;
+
+pub type StoppedClockExtentMaker = ExtentClock<{ ClockType::StoppedClock as usize }>;
+
+impl ExtentFromClock<WorkingClock> for WorkingClockExtentMaker {}
+
+impl ExtentFromClock<StoppedClock> for StoppedClockExtentMaker {}
+
+#[cfg(not(test))]
+pub type DefaultClockExtentMaker = WorkingClockExtentMaker;
+
+#[cfg(test)]
+pub type DefaultClockExtentMaker = StoppedClockExtentMaker;
```

Comment:
@da2ce7 should not be `DefaultExtendClockMaker`? Because you are making an EntentClock.

Reply:
I've updated the names: `DefaultClockTimeExtentMaker` is uses the default clock for the `TimeExtentMaker` implementation.